src/ClassDiscovery.php

```diff
@@ -2,126 +2,117 @@
 
 namespace Http\Discovery;
 
-use Puli\Discovery\Api\Discovery;
+use Http\Discovery\Exception\DiscoveryFailedException;
+use Http\Discovery\Exception\StrategyUnavailableException;
 
 /**
  * Registry that based find results on class existence.
  *
  * @author David de Boer <david@ddeboer.nl>
  * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
  */
 abstract class ClassDiscovery
 {
     /**
-     * @var GeneratedPuliFactory
+     * A list of strategies to find classes.
+     *
+     * @var array
      */
-    private static $puliFactory;
+    public static $strategies = [
```

Comment:
Yes, we are using strategies that we know would fail. Example; the HttpClients strategy will always fail for the FactoryDiscovery. I'll fix this by merging the strategies suggested bellow.